src/test_helper.rs

```diff
@@ -0,0 +1,213 @@
+        });
+        event_tx_clone.blocking_send(app.get_event_tx()).unwrap();
+
+        app.get_terminal()
+            .backend_mut()
+            .resize(buffer_size.width, buffer_size.height);
+
+        app.run()
+    });
+
+    let event_tx = event_rx.blocking_recv().unwrap();
+
+    let buffer = message_rx.blocking_recv().unwrap();
+
+    event_tx.blocking_send(Event::Quit).unwrap();
+    let _ = join_handle.join().unwrap();
+    buffer
+}
+
+/// Render a widget and return the terminal to check the generated output
+///
+/// * `buffer_size` - The terminal output buffer is set to that size.
+/// * `sut` - (system under test) The widget to render.
+pub fn render_widget(buffer_size: Size, sut: &mut impl Widget) -> Terminal<TestBackend> {
+    let mut messages = vec![];
+    let mut cx_state = CxState::new(&mut messages);
+    let mut widget_state = WidgetState::new();
+
+    let mut layout_cx = LayoutCx {
+        cx_state: &mut cx_state,
+        widget_state: &mut widget_state,
+    };
+    let backend = TestBackend::new(buffer_size.width, buffer_size.height);
+
+    let mut terminal = Terminal::new(backend).unwrap();
+    let _size = sut.layout(
+        &mut layout_cx,
+        &BoxConstraints::new(
+            kurbo::Size::ZERO,
+            kurbo::Size {
+                width: buffer_size.width.into(),
+                height: buffer_size.height.into(),
+            },
+        ),
+    );
+
+    let mut paint_cx = PaintCx {
+        cx_state: &mut cx_state,
+        widget_state: &mut widget_state,
+        terminal: &mut terminal,
+        override_style: Style::default(),
+    };
+
+    sut.paint(&mut paint_cx);
+    terminal.flush().unwrap();
+
+    terminal
+}
+
+/// This widget provides access to the terminal output of its children
+///
+/// After its children were painted it calls the flush() and clones the
+/// terminal's buffer and sends it using the passed [Sender<Buffer>].
+///
+/// This is handy for snapshot tests.
+pub struct DebugView<V, T, A> {
+    content: V,
+    debug_chan_tx: mpsc::Sender<Buffer>,
+    phantom: PhantomData<fn() -> (T, A)>,
+}
+
+pub fn debug_view<V, T, A>(content: V, debug_chan_tx: mpsc::Sender<Buffer>) -> DebugView<V, T, A> {
+    DebugView {
+        content,
+        debug_chan_tx,
+        phantom: PhantomData,
+    }
+}
+
+impl<T, A, V> ViewMarker for DebugView<V, T, A> {}
+
+impl<T, A, V> View<T, A> for DebugView<V, T, A>
+where
+    V: View<T, A>,
+    V::Element: 'static,
+{
+    type State = (V::State, xilem_core::Id);
+
+    type Element = DebugWidget;
+
+    fn build(&self, cx: &mut Cx) -> (xilem_core::Id, Self::State, Self::Element) {
+        let (id, (state, element)) = cx.with_new_id(|cx| {
+            let (child_id, state, element) = self.content.build(cx);
+            (
+                (state, child_id),
+                DebugWidget::new(element, self.debug_chan_tx.clone()),
+            )
+        });
+        (id, state, element)
```

Comment:
Since this view doesn't take any messages directly, it's probably simpler to not use a separate id at all (needs some adjustments in `View::State` and the other methods as well)
```suggestion
        let (id, state, element) = self.content.build(cx);
        (id, state, DebugWidget::new(element, self.debug_chan_tx.clone()))
```